cg/models/demultiplex/flow_cell.py

```diff
@@ -75,21 +81,28 @@ def full_name(self) -> str:
         """Return flow cell full name."""
         return self.path.name
 
+    @property
+    def sample_sheet(self) -> SampleSheet:
+        """Return sample sheet object."""
+        if not self._sample_sheet:
+            self._sample_sheet = self._read_sample_sheet()
+        return self._sample_sheet
+
+    def _read_sample_sheet(self) -> SampleSheet:
+        if not self.sample_sheet_path_hk:
+            raise FlowCellError("Sample sheet path not set")
+        return get_sample_sheet_from_file(
+            infile=self.sample_sheet_path_hk,
+            flow_cell_sample_type=self.sample_type,
+        )
```

Comment:
It might make more sense to pass the actual `SampleSheet` object to the constructor of the `FlowCellDirectoryData` class instead. I think the class becomes easier to test that way and it would not contain logic for parsing a sample sheet any longer

Reply:
I think that would move the reading/ parsing of the sample sheet to other places where we call the `FlowCellDirectoryData`? Not sure if that would require more changes overall.

Reply:
I don't think it would since it is an optional argument and we only provide it in the post processing flow

Reply:
We can decide on it later